lib/elixir/src/elixir_rewrite.erl

```diff
@@ -302,8 +302,6 @@ increment(Meta, Other) ->
 %% The allowed operations are very limited.
 %% The Kernel operators are already inlined by now, we only need to
 %% care about Erlang ones.
-match_rewrite(erlang, _, '+', _, [Arg]) when is_number(Arg) -> {ok, Arg};
-match_rewrite(erlang, _, '-', _, [Arg]) when is_number(Arg) -> {ok, -Arg};
 match_rewrite(erlang, _, '++', Meta, [Left, Right]) ->
```

Comment:
Can you please do a check later that we still need this one? I think recent Erlang versions started inlining `++` as well. :) It may be we can get rid of `match_rewrite` altogether!

Reply:
Sounds good, will definitely check!

Reply:
OK, I tried for a while but I don't think we can reasonably remove it without introducing regressions.

### Issue 1: Elixir's current inlining is more powerful

```elixir
iex(5)> [a] ++ bc = 'abc'
~c"abc"
```

```erlang
1> [A] ++ BC = "abc".
* 1:5: illegal pattern

2> "a" ++ BC = "abc".
"abc"
```

### Issue 2: Rewrite currently serves as validation, will need to translate more erlang errors

Currently:

```elixir
iex(1)> "a" ++ "b" = "ab"
error: invalid argument for ++ operator inside a match, expected a literal proper list, got: "a"
```

Relying on erlang's inlining:

```elixir
iex(1)> "a" ++ "b" = "ab"
** (ErlangError) Erlang error: {:illegal_pattern, {:op, 1, :++, {:bin, 1, [{:bin_element, 1, {:string, 1, ~c"a"}, :default, :default}]}, {:bin, 1, [{:bin_element, 1, {:string, 1, ~c"b"}, :default, :default}]}}}
```

Reply:
Thank you for checking!